File: Easy/349-intersection-of-two-arrays/try-8.py
/*
 * Problem #349: Intersection of Two Arrays
 * Difficulty: Easy
 * Submission: Try 8
 * status: Accepted
 * Language: python3
 * Date: 06/12/2024, 15:10:50
 * Link: https://leetcode.com/problems/intersection-of-two-arrays/
 */

class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        ans = []
        for value in nums1:
            if value in nums2:
                ans.append(value)
        for value in nums2:
            if value in nums1:
                ans.append(value)
        temp = []
        for value in ans:
            if value not in temp:
                temp.append(value)
        
        return temp

        return list(num_ans)
        

File: Easy/350-intersection-of-two-arrays-ii/try-8.cpp
/*
 * Problem #350: Intersection of Two Arrays II
 * Difficulty: Easy
 * Submission: Try 8
 * status: Accepted
 * Language: cpp
 * Date: 16/01/2025, 17:00:00
 * Link: https://leetcode.com/problems/intersection-of-two-arrays-ii/
 */

class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        unordered_map<int,int> map;
        vector<int> vec;
        for(auto i : nums1){
            map[i]++;
        }
        for(auto j : nums2){
            if(map[j]>0){
                vec.push_back(j);
                map[j]--;
            }
        }
        return vec;
    }
};
